### ISSUE TITLE
Command Palette: add command to open fonts

### ISSUE THREAD

**Issue by annezazu** [WordPress/gutenberg#54880]
Pulled from the [FSE Outreach Program's Final Touches call for testing](https://make.wordpress.org/test/2023/09/18/fse-program-testing-call-26-final-touches/#comment-3177):

>I clicked the “Aa” icon to open up font management just because I followed your instructions, but I couldn’t have found the font management easily without them. I love icons, but this is only visible with a label. I guess this is by design. I can’t wait for a font’s library command.

In checking, there is not a command in place for opening the font management and it seems wise to add in. At the minimum, a "Open font management" command, similar to "Open styles".

<img width="547" alt="Screenshot 2023-09-27 at 10 03 17 AM" src="https://github.com/WordPress/gutenberg/assets/26996883/b1860aba-b30a-4a77-be50-b59c1391c63c">

**Comment by richtabor** [comment]
I would expect this command would available in the Site Editor, opening the Styles > Typography panel, with the Fonts modal open. 

<img width="1237" alt="CleanShot 2023-09-28 at 12 50 46" src="https://github.com/WordPress/gutenberg/assets/1813435/6c2bf782-0767-4450-96bc-f1dd7a57a321">

 <img width="480" alt="CleanShot 2023-09-28 at 12 49 25" src="https://github.com/WordPress/gutenberg/assets/1813435/47de37e6-e651-48ee-89a6-434def585fa3">

**Comment by jordesign** [comment]
@annezazu It strikes me that if the Font Library is going to land in 6.5 - that this would be a good one to squeeze in as well.

**Comment by annezazu** [comment]
Agreed! @mikachan & @matiasbenedetto for consideration.

**Comment by matiasbenedetto** [comment]
Thanks for the ping. It would be very nice to add this :+1:

**Comment by mikachan** [comment]
Agreed, this would be a super cool addition!

**Comment by carolinan** [comment]
I had a try at this, but was unable to find a way to open not only the modal, but also the typography panel.

In comparison, (simplified) the custom CSS, stylebook and preference modal commands work because these are separate screens with their own paths.

**Comment by mikachan** [comment]
Thanks for taking a look at this, @carolinan. Maybe we need to use a different kind of trigger to open the Font Library modal, something that's more similar to the preference modal. (I say all this without looking into the details yet.)

**Comment by t-hamano** [comment]
Just to be sure, is this command expected to open the Font Library modal itself, rather than opening the Global Styles typography panel? If so, I think we need to manage the modal's state with a datastore. I'm testing it with #58350.

**Comment by mikachan** [comment]
>  is this command expected to open the Font Library modal itself, rather than opening the Global Styles typography panel?

Yes, I think opening the modal makes the most sense. Thanks for testing a solution for this! Managing the modal's state with the datastore sounds like a good step forward.

**Comment by carolinan** [comment]
I think it should do both.

**Comment by t-hamano** [comment]
> I think it should do both.

After some simple hacking and testing of existing command actions, it seems possible to open the Typograph panel with a command action.

I would like to proceed with #58350.

<details><summary>Diff</summary>

```diff
diff --git a/packages/edit-site/src/components/global-styles/ui.js b/packages/edit-site/src/components/global-styles/ui.js
index 3abd1c811f..c59c7caee3 100644
--- a/packages/edit-site/src/components/global-styles/ui.js
+++ b/packages/edit-site/src/components/global-styles/ui.js
@@ -246,6 +246,9 @@ function GlobalStylesEditorCanvasContainerLink() {
                        case 'global-styles-revisions:style-book':
                                goTo( '/revisions' );
                                break;
+                       case 'global-styles-typograpy':
+                               goTo( '/typography' );
+                               break;
                        case 'global-styles-css':
                                goTo( '/css' );
                                break;
diff --git a/packages/edit-site/src/hooks/commands/use-common-commands.js b/packages/edit-site/src/hooks/commands/use-common-commands.js
index fdea50d8af..e61365572d 100644
--- a/packages/edit-site/src/hooks/commands/use-common-commands.js
+++ b/packages/edit-site/src/hooks/commands/use-common-commands.js
@@ -202,8 +202,8 @@ function useGlobalStylesOpenCssCommands() {
 
                return [
                        {
-                               name: 'core/edit-site/open-styles-css',
-                               label: __( 'Customize CSS' ),
+                               name: 'core/edit-site/open-font-library-modal',
+                               label: __( 'Open font library' ),
                                icon: brush,
                                callback: ( { close } ) => {
                                        close();
@@ -217,7 +217,7 @@ function useGlobalStylesOpenCssCommands() {
                                                setCanvasMode( 'edit' );
                                        }
                                        openGeneralSidebar( 'edit-site/global-styles' );
-                                       setEditorCanvasContainerView( 'global-styles-css' );
+                                       setEditorCanvasContainerView( 'global-styles-typograpy' );
                                },
                        },
                ];
```

</details> 

https://github.com/WordPress/gutenberg/assets/54422211/81b76803-2979-47ea-9ca6-07a8d6d474ae

**Comment by richtabor** [comment]
> I think it should do both.

Agreed.

**Comment by t-hamano** [comment]
As seen in https://github.com/WordPress/gutenberg/pull/58350#discussion_r1470690520, refactoring (removing) the context is necessary to move this issue forward.

I have submitted https://github.com/WordPress/gutenberg/issues/58428, but if anyone is interested, please join.

**Comment by t-hamano** [comment]
In order to add the command using the ideal approach, #58428 would need to be resolved first. This command is not originally on the roadmap and I don't think it's essential to include it in WP6.5, so I'll punt it to 6.6.

**Comment by NekoJonez** [comment]
Today I was playing around looking for the modal in 6.5 beta 2 ... And I didn't easily find it. Now, does this mean that the font library is only here for FSE themes? Is it ever coming to non FSE themes?

Also, I expected it to be in the block options but don't know if that's the best idea.

**Comment by colorful-tones** [comment]
Hi folks,
We are only one week away from the Beta 1 cut-off date for [WordPress 6.6](https://make.wordpress.org/core/6-6/). This issue hasn’t seen any movement in a while, so we (the editor triage leads of the 6.6 release) have decided to remove it from the [WordPress 6.6 Editor Tasks project board](https://github.com/orgs/WordPress/projects/179/views/1).

**Comment by annezazu** [comment]
Closing this out as there's now a dedicated font view and this opens correctly when searched for in the command palette:

https://github.com/user-attachments/assets/88514a4c-128b-4194-a9c7-fd0695efc9f0
